location/index.js

```diff
@@ -84,145 +132,162 @@ class IRNMNLocation extends HTMLElement {
     }
 
     /**
-     * Get the placeholder for the location select.
+     * Gets the placeholder for the location dropdown.
      *
-     * @return {String} Placeholder or default value 'Select a location'.
+     * @return {string} The placeholder text or a default value.
      */
     get placeholder() {
         return this.getAttribute('placeholder') || 'Select a location';
     }
 
-    /**
-     * Get the show error flag for the location select.
-     * @return {Boolean} Show error flag or default value false.
-     */
     get showError() {
         return (
             this.hasAttribute('show-error') &&
             this.getAttribute('show-error') === 'true'
         );
     }
 
-    /**
-     * Get the error message for the location select.
-     *
-     * @return {String} Error message or default value 'Please select a valid location'.
-     */
     get errorMessage() {
         return (
             this.getAttribute('error-message') ||
             'Please select a valid location'
         );
     }
 
-    /**
-     * Render the component only if locations data is available.
-     */
     render() {
         if (!this.locations || this.locations.length === 0) {
             console.error('No locations provided');
             return;
         }
-
+    
+        const options = this.locations.map((location) => ({
+            value: location.hotelCode, // Used as the unique identifier
+            name: location.hotelName, // Used as the display text
+        }));
+    
+        // Get the hidden input value or default value
+        const preselectedValue = getFromSessionStorage(this.inputName) || this.default || '';
+    
         this.innerHTML = `
             <div class="${CLASS_NAMES.container}">
                 <label for="${this.inputId}" class="${CLASS_NAMES.label}">${this.label}</label>
-                <select id="${this.inputId}" name="${this.inputName}" class="${CLASS_NAMES.select}" required>
-                    <option value="" disabled selected>${this.placeholder}</option>
-                    ${this.locations
-                        .map((location) => {
-                            // Dynamically create the data attributes based on the locations obj
-                            const dataAttributes = Object.entries(location)
-                                .map(([key, value]) => {
-                                    const dataAttrName = `data-${key.replace(/([A-Z])/g, '-$1').toLowerCase()}`;
-                                    return `${dataAttrName}="${value}"`;
-                                })
-                                .join(' ');
-
-                            return `
-                            <option value="${location.hotelCode}" ${dataAttributes} class="${CLASS_NAMES.option}">
-                                ${location.hotelName}
-                            </option>`;
-                        })
-                        .join('')}
-                </select>
+                <irnmn-select 
```

Comment:
@ejcromwell here you can see <irnmn-select being now used in the location dropdown, instead of the native one. 
Also, we have a hidden output that represents the hotelCode